src/lib.rs

```diff
@@ -834,29 +834,33 @@ where
     }
 }
 
+fn arbitrary_str<'a>(u: &mut Unstructured<'a>, size: usize) -> Result<&'a str> {
+    match str::from_utf8(u.peek_bytes(size).unwrap()) {
+        Ok(s) => {
+            u.bytes(size).unwrap();
+            Ok(s)
+        }
+        Err(e) => {
+            let i = e.valid_up_to();
+            let valid = u.bytes(i).unwrap();
+            let s = unsafe {
+                debug_assert!(str::from_utf8(valid).is_ok());
```

Comment:
thought: should we be including this? fuzzing is usually run with debug assertions on so the debug_assert doesn't get us much. I am comfortable with more thoroughly documenting the safety situation instead

Reply:
We have a couple other debug asserts that also maybe should only be enabled when running the `arbitrary` crate tests. We can figure out what to do with all of them in a follow up, no need to hold this PR up.

Reply:
This `debug_assert` comes from the previous implementation of `arbitrary`; if you think it’d be better to replace it with a comment that `valid_up_to` can only generate valid strings, or with another macro that’d assert only on eg. an `internal-tests` arbitrary feature, I’d be happy to make a follow-up PR :)